src/Regexp.h

```diff
@@ -0,0 +1,131 @@
+/*
+ 
+ Regular-expression matching library for Arduino.
+ 
+ Written by Nick Gammon.
+ Date: 30 April 2011
+ 
+ Heavily based on the Lua regular expression matching library written by Roberto Ierusalimschy. 
+ 
+ Adapted to run on the Arduino by Nick Gammon.
+ 
+ 
+ VERSION
+ 
+  Version 1.0  - 30th April 2011 : initial release.
+  Version 1.1  - 1st May 2011    : added some helper functions, made more modular.
+  Version 1.2  - 19th May 2011   : added more helper functions for replacing etc.
+
+
+ */
+
+#pragma once
```

Comment:
This is, by the way, the one little change I made to this file to prevent problems when your library is included in multiple places in a complex project.